sync/class.jetpack-sync-module-callables.php

```diff
@@ -23,12 +23,19 @@ public function set_defaults() {
 	public function init_listeners( $callable ) {
 		add_action( 'jetpack_sync_callable', $callable, 10, 2 );
 
-		// always send change to active modules right away
-		add_action( 'update_option_jetpack_active_modules', array( $this, 'unlock_sync_callable' ) );
+		// For some options, we should always send the change right away!
+		$always_send_updates_to_these_options = array( 'jetpack_active_modules', 'home', 'siteurl' );
+		foreach( $always_send_updates_to_these_options as $option ) {
+			add_action( "update_option_{$option}", array( $this, 'unlock_sync_callable' ) );
+		}
 
 		// get_plugins and wp_version
 		// gets fired when new code gets installed, updates etc.
 		add_action( 'upgrader_process_complete', array( $this, 'unlock_sync_callable' ) );
+
+		// Provide a hook so that hosts can send changes to certain callables right away.
+		// Especially useful when a host uses constants to change home and siteurl.
+		add_action( 'jetpack_sync_unlock_sync_callable', array( $this, 'unlock_sync_callable' ) );
```

Comment:
I think this part is not necessery Since the dev could just call the public method right `unlock_sync_callable` away.

Currently we go and calculate these values right away anyway right after the transient expires. Which might not be fast enough works.